src/testnet/run_loop.rs

```diff
@@ -14,7 +14,7 @@ pub struct RunLoop {
     nodes: Vec<Node>,
     new_burnchain_state_callback: Option<fn(u8, &BurnchainState)>,
     new_tenure_callback: Option<fn(u8, &LeaderTenure)>,
-    new_chain_state_callback: Option<fn(u8, &StacksChainState)>,
+    new_chain_state_callback: Option<fn(u8, &mut StacksChainState, &BlockHeaderHash)>,
```

Comment:
Not saying that we have to do this in this particular PR, but I'm increasingly uncomfortable with these being a set of callbacks that may or may not be `Some(...)`, since it introduces the need for a lot of runtime boiilerplate checks.  Can either you or @lgalabru open an issue to make `RunLoop` a `trait` instead, and then refactor this code into one or more `impl`s of that trait with the requisite behaviors?  I can see there being at least two `impl`s for example -- one for a private "regtest" and one for a public "testnet".

Reply:
I don't generally think its a good idea to design a trait before there are actually multiple implementations of the trait. Otherwise, we end up thinking we know what the common interface should be before actually knowing what it should be.

For the boiler plate checks on these callbacks, I can just refactor these checks into function calls that handle the Option matching.

Reply:
Sounds good.  I'm fine with deferring making this a trait until we have two non-trivial impls, but I'd nevertheless appreciate moving the Option matching to one place for the time being.

Reply:
Addressed in d43618a12c87932389a5e41ec3a575bc6e342f8d